system/mp/mpcomm.cpp

```diff
@@ -900,15 +932,35 @@ protected: friend class CMPPacketReader;
 #ifdef _FULLTRACE
                     PROGLOG("MP: rd = %d", rd);
 #endif
-                    if (rd != 0)
+                    /* NB: legacy clients that don't handle the exception deserialization here
+                     * will see reply as success, so no clean error,
+                     * but will fail shortly afterwards since server connection is closed
+                     */
+                    if (rd > sizeof(rd)) // legacy clients will only ever send a reply of 0 or 4, if greater, then new client is replying with an exception
+                    {
+                        MemoryBuffer mb;
+                        mb.setBuffer(rd, replyMem, false);
+                        size32_t len;
+                        mb.read(len); // exception length
+                        if (len)
```

Comment:
Doesn't really matter, but I don't think len can ever be null - e.g., a null exception would add a single byte to the output.

Reply:
Yeah I knew it couldn't, but I added check deliberately for possible future backward compatibility, e.g. no exception sent, but reply extended to send something else.